### PR TITLE
Ability to use logger feature of gitonomy/gitlib

### PR DESCRIPTION
**Pull request by llaville**
Use console verbose level 1, 2, or 3

Examples:
```
grumphp run -v
[2016-02-06 13:23:40] RegisteredFiles.INFO: run command: ls-files ""  [] []
[2016-02-06 13:23:41] RegisteredFiles.DEBUG: last command (ls-files) duration: 312.00ms [] []
[2016-02-06 13:23:41] RegisteredFiles.DEBUG: last command (ls-files) return code: 0 [] []
[2016-02-06 13:23:41] RegisteredFiles.DEBUG: last command (ls-files) output: task/Phpparser.php task/sample2.php  [] []
GrumPHP is sniffing your code!
Running task 1/6: Phpcs
Running task 2/6: Phpspec
Running task 3/6: Phpunit
Running task 4/6: Composer
Running task 5/6: SecurityChecker
Running task 6/6: YamlLint
```

```
grumphp git:pre-commit -vv
[2016-02-06 13:26:11] ChangedFiles.INFO: run command: diff "-r -p -m -M --full-index --staged"  [] []
[2016-02-06 13:26:11] ChangedFiles.DEBUG: last command (diff) duration: 327.60ms [] []
[2016-02-06 13:26:11] ChangedFiles.DEBUG: last command (diff) return code: 0 [] []

and more ...
```

```
grumphp git:commit-msg -v "blah bla"
[2016-02-06 13:27:43] ChangedFiles.INFO: run command: diff "-r -p -m -M --full-index --staged"  [] []
[2016-02-06 13:27:43] ChangedFiles.DEBUG: last command (diff) duration: 327.60ms [] []
[2016-02-06 13:27:43] ChangedFiles.DEBUG: last command (diff) return code: 0 [] []

and lot more ...
```

### CODE REVIEW

**Comment by veewee** [comment]
Maybe it's easier to specify 1 logger in the configuration.
Instead of adding the interface to the locators, the setLogger() method is directly called on the Repository object in the commands.
If the command is not verbose, you just set a NullHandler as Log handler.

The repository is also used in some tasks / listeners, so the logger is not only attached to the locator.

Pretty nice feature though :)

File: src/GrumPHP/Locator/AbstractFiles.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Sets a PSR-3 logger
+     * @return $this
+     */
+    public function setLogger()
+    {
+        $this->repository->setLogger($this->logger);
+
+        return $this;
```
Comment (aderuwe) [inline src/GrumPHP/Locator/AbstractFiles.php]: Please don't do this, fluent interfaces make sense only in some cases.

**Comment by veewee** [comment]
@llaville,

I created a new PR for the logger in #114.
This one creates a bit less overhead and makes it easy to configure the logger for external tasks.

**Comment by llaville** [comment]
@veewee Will have a investigate deeply in code in few hours. But look nice in first look :)

**Pull request closed without merging**